transport/server.cc

```diff
@@ -238,14 +238,21 @@ cql_server::do_accepts(int which, bool keepalive, socket_address server_addr) {
             auto conn = make_shared<connection>(*this, server_addr, std::move(fd), std::move(addr));
             ++_connects;
             ++_connections;
-            // Move connection to the background, monitor for lifetime and errors.
-            (void)conn->process().then_wrapped([this, conn] (future<> f) {
-                --_connections;
+            // Move conn to the background, announce, then block while monitoring for lifetime/errors.
+            (void)futurize_apply([this, conn] {
+                return advertise_new_connection(conn);
+            }).then_wrapped([this, conn] (future<> f) {
                 try {
                     f.get();
-                } catch (...) {
-                    clogger.debug("connection error: {}", std::current_exception());
+                } catch(...) {
+                    clogger.debug("exception while advertising new connection: {}", std::current_exception());
                 }
```

Comment:
Should be at INFO level, or we'll never see it in regular deployments. Since we don't expect an exception under normal circumstances, it won't pollute the log too much.

Reply:
Done.